web/styles/zulip.css

```diff
@@ -3295,3 +3295,86 @@ select.invite-as {
         padding: 5px;
     }
 }
+
+#add-poll-modal {
+    /* on wide screens, this height allows roughly 3 option
+    rows to fit in without need for scrolling */
+    height: 450px;
+    overflow: hidden;
+
+    @media (width < $sm_min) {
+        height: 80vh;
+    }
+
+    .modal__content {
+        flex-grow: 1;
+
+        .simplebar-content {
+            box-sizing: border-box;
+            height: 100%;
+            display: flex;
+            flex-direction: column;
+        }
+    }
+
+    #add-poll-form {
+        display: flex;
+        flex-direction: column;
+        overflow: hidden;
+        height: 100%;
+
+        & label {
+            font-weight: bold;
+            margin: 5px 0;
+        }
+
+        & input[type="text"] {
+            border: 1px solid hsl(0deg 0% 0% / 20%);
+            outline: 0;
+            height: 1.8rem;
+        }
+
+        #poll-question-input {
+            margin-bottom: 10px;
+        }
+
+        .poll-options-list {
+            margin: 0;
+            height: 0;
+            overflow: auto;
+            flex-grow: 1;
+
+            .option-row {
```

Comment:
Could the option rows be made into flexboxes? That might reduce some of the complexity here with various bits of margin, or the relative positioning on the elipsis.

Reply:
Thanks for the idea! This did reduce the complexity, and options (and the question input) taking up all available space looks fine too